python/ray/data/_internal/progress_bar.py

```diff
@@ -105,11 +107,17 @@ def set_description(self, name: str) -> None:
             self._bar.set_description(self._desc)
 
     def update(self, i: int) -> None:
+        # Make sure the progress bar doesn't go over 100%.
+        i = min(i, self._total - self._position)
```

Comment:
I think we do want to show it going over 100%, otherwise it will look "stuck" at 100%, right? Maybe in that case we should change the total to be higher?

Reply:
ah right, I think we could change `self._total` to reflect the progress.

Reply:
I think you might have to also update self._bar's total somehow. I'm not sure about the tqdm API here, might need to try it out.

Reply:
If not possible, going over 100% is OK in tqdm as well, it just goes from a progress bar to a plain meter when you exceed 100.

Reply:
I didn't find an API, but this seems to work well: `bar.total = new_total; bar.refresh()`.
However, even if we update the total, the bar will still stuck at 100%, right? The only benefit is the progress value will change.
Do you guys prefer this?

Reply:
By the way, do you know what is the simplest way to reproduce the over 100% case?

Reply:
I end up updating the total. I think this is useful, as we can see the actual number of blocks from here.